regression/goto-analyzer-simplify/simplify-complex-expression/test.desc

```diff
@@ -0,0 +1,5 @@
+CORE
+main.c
+"--variable"
+
+r == 2
```

Comment:
What is this test supposed to test for? That is, why is `r == 2` the one and only crucial output to occur?

Reply:
This is testing for an earlier bug in which there was an incorrect replacement of symbols by the RHS of their last assignment.

Reply:
Would you mind adding this as a comment, possibly also referring to the commit that fixed the bug? (You can add free-text comments after two `---` lines.)

Reply:
This bug was caused by an earlier version of this commit (that didn't have the part below for the for loop) and is fixed with the current version. I left it in so I wouldn't accidentally break it again.

Reply:
Ok, great, please just include some explanation.